### PR TITLE
[CC] x-forward-for header manipulation using `use_remote_address` router config

### PR DESCRIPTION
**Pull request by NomadXD**
## Purpose
$subject

### CODE REVIEW

**Comment by NomadXD** [comment]
Related menu/sub-menu needs to be changed according to https://github.com/wso2/product-microgateway/issues/2969

**Review by pubudu538 — APPROVED** [review]

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/configurations/router-configurations.md [hunk 1]
```diff
@@ -196,6 +197,25 @@ See the example .toml file given below.
...
+                                            <span class="param-type string"> boolean </span>
+                                            
+                                        </p>
+                                        <div class="param-default">
+                                            <span class="param-default-value">Default: <code>false</code></span>
+                                        </div>
+                                        
+                                    </div>
+                                    <div class="param-description">
+                                        <p>If configured true, router appends the immediate downstream IP address to the x-forward-for header.</p>
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/configurations/router-configurations.md:216]:
```suggestion
                                        <p>If configured as true, the Router appends the immediate downstream IP address to the x-forward-for header.</p>
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/howtos/xff-header.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+x-forwarded-for (XFF) is a standard proxy header defined by [RFC 7239](https://datatracker.ietf.org/doc/html/rfc7239.html) which indicates the IP addresses that a request has flowed through on its way from the client to the server. Choreo Connect supports x-forward-for header manipulation using `use_remote_address` field in the router configuration.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/howtos/xff-header.md:1]: Please add a proper title for this page.

```suggestion
x-forwarded-for (XFF) is a standard proxy header defined by [RFC 7239](https://datatracker.ietf.org/doc/html/rfc7239.html), which indicates the IP addresses that a request has flowed through on its way from the client to the server. Choreo Connect supports x-forward-for header manipulation using `use_remote_address` field in the router configuration.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/howtos/xff-header.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+x-forwarded-for (XFF) is a standard proxy header defined by [RFC 7239](https://datatracker.ietf.org/doc/html/rfc7239.html), which indicates the IP addresses that a request has flowed through on its way from the client to the server. Choreo Connect supports x-forward-for header manipulation using `use_remote_address` field in the Router configuration.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/howtos/xff-header.md:1]: Please add a proper title for this page.

File: en/tools/config-catalog-generator-cc/dist/router-configurations.md [hunk 4]
```diff
@@ -196,6 +197,25 @@ See the example .toml file given below.
...
+                                            <span class="param-type string"> boolean </span>
+                                            
+                                        </p>
+                                        <div class="param-default">
+                                            <span class="param-default-value">Default: <code>false</code></span>
+                                        </div>
+                                        
+                                    </div>
+                                    <div class="param-description">
+                                        <p>If configured true, router appends the immediate downstream IP address to the x-forward-for header.</p>
```
Comment (samgnaniah) [inline en/tools/config-catalog-generator-cc/dist/router-configurations.md:216]:
```suggestion
                                        <p>If configured as true, the Router appends the immediate downstream IP address to the x-forward-for header.</p>
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/xff-header.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# `x-forwarded-for` header manipulation
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/xff-header.md:1]:
```suggestion
# XFF Header Manipulation
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/xff-header.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# XFF Header Manipulation
+
+x-forwarded-for (XFF) is a standard proxy header defined by [RFC 7239](https://datatracker.ietf.org/doc/html/rfc7239.html), which indicates the IP addresses that a request has flowed through on its way from the client to the server. Choreo Connect supports x-forwarded-for header manipulation using `use_remote_address` field in the Router configuration.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/xff-header.md:3]:
```suggestion
`x-forwarded-for` (XFF) is a standard proxy header defined by [RFC 7239](https://datatracker.ietf.org/doc/html/rfc7239.html), which indicates the IP addresses that a request has flowed through on its way from the client to the server. Choreo Connect supports x-forwarded-for header manipulation using `use_remote_address` field in the Router configuration.
```

**Review by samgnaniah — APPROVED** [review]

**Pull request merged**